spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraConfSetting.scala

```diff
@@ -0,0 +1,190 @@
+package org.apache.spark.sql.cassandra
+
+import com.datastax.spark.connector.cql.CassandraConnectorConf
+import org.apache.commons.lang.StringUtils
+
+/**
+ * A local cache to store user defined Cassandra connection configuration settings. It includes
+ * per cluster level, per keyspace level and per table level settings. It also allows to
+ * update the settings.
+ */
+class CassandraConnConfSetting extends CassandraConfSetting[CassandraConnectorConf] {
+
+  import CassandraConfSetting._
+
+  /** Get Cassandra connection configuration settings for a cluster */
+  def getConf(
+      cluster: Option[String],
+      defaultConf: CassandraConnectorConf): CassandraConnectorConf = {
+
+    cluster match {
+      case Some(c) => validateClusterName(c)
+        Option(clusterConf.get(c)).getOrElse(
+          throw new RuntimeException(s"Missing cluster $c Cassandra connection conf"))
+      case _ => defaultConf
+    }
+  }
+
+  override def addTableConf(
+               table: String,
+               keyspace: String,
+               cluster: Option[String],
+               conf: CassandraConnectorConf): Unit = {
+
+    throw new UnsupportedOperationException("Add table level CassandraConnectorConf is not supported")
```

Comment:
If an operation doesn't make any sense, it should not be a part of the class API.